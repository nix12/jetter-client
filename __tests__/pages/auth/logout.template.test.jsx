import React from 'react';
import { shallow } from 'enzyme';
import Template from '../../../templates/auth/logout.template';

describe('<Logout /> template', () => {
  it('should render Logout component', () => {
    const props = {
      onLogout: jest.fn()
    };
    const wrapper = shallow(<Template {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should logout and redirect', () => {
    jest.useFakeTimers();

    const props = {
      onLogout: jest.fn(),
      router: {
        push: jest.fn()
      }
    };

    const wrapper = shallow(<Template {...props} />);
    const instance = wrapper.instance();
    instance.componentDidMount();

    expect(setTimeout).toHaveBeenCalledTimes(2);

    jest.runAllTimers();
    expect(instance.props.onLogout).toBeCalled();
  });
});
