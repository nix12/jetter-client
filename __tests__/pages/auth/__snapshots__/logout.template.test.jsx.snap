// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logout /> template should render Logout component 1`] = `ShallowWrapper {}`;
